video/out/gpu/d3d11_helpers.c

```diff
@@ -85,7 +146,7 @@ static HRESULT create_device(struct mp_log *log, bool warp, bool debug,
     D3D_DRIVER_TYPE type = warp ? D3D_DRIVER_TYPE_WARP
                                 : D3D_DRIVER_TYPE_HARDWARE;
```

Comment:
This isn't working because the DriverType parameter should be D3D_DRIVER_TYPE_UNKNOWN when pAdapter as non-NULL. Try this:

```c
    D3D_DRIVER_TYPE type = D3D_DRIVER_TYPE_UNKNOWN;
    if (!adapter) {
        if (warp) {
            type = D3D_DRIVER_TYPE_WARP;
        } else {
            type = D3D_DRIVER_TYPE_HARDWARE;
        }
    }
```

Reply:
Right. I did guess it was something atrociously dumb like that which I had forgotten :) .

It did get rather late.